docs/getting-started-guides/scratch.md

```diff
@@ -0,0 +1,664 @@
+bare-metal), and not all parts of the interface need to be implemented, depending
+on how flags are set on various components. 
+
+### Selecting Nodes
+- You can use virtual or physical machines.
+- The machines need to be on the same subnet.
+- While you can build a cluster with 1 machine, in order to run all the examples and tests you
+  need at least 4 nodes.
+- Many Getting-started-guides make a distinction between the master node and regular nodes.  This
+  is not strictly necessary.
+- Nodes will need to run some version of Linux with the x86_64 architecture.  It may be possible
+  to run on other OSes and Architectures, but this guide does not try to assist with that.
+- Apiserver and etcd together are fine on a machine with 1 core and 1GB RAM for clusters with 10s of nodes.
+  Larger or more active clusters may benefit from a more cores.
+- Other nodes can have any reasonable amount of memory and any number of cores.  They need not
+  have identical configurations.
+
+### Selecting a Network
+Kubernetes has a distinctive [networking model](../networking.md) which uses
+"virtual" IPs for pods and services and a different virtual ethernet device for each pod.
+
+There are broadly speaking two approaches to configuring in-cluster networking:
+- Create an Overlay network
+  - Easier to setup
+  - Traffic is encapsulated, so per-pod virtual IPs are routable.
+  - Examples:
+    - [Flannel](https://github.com/coreos/flannel) 
+    - [Weave](http://weave.works/)
+  - Does not require "Routes" portion of Cloud Provider module.
+- Configure network to route Pod IPs
+  - Harder to setup from scratch.
+  - Implemented for GCE and AWS.
+  - Uses Cloud Provider modules with "Routes" subinterface.
+  - Theoretically possible to implement for bare-metal networking, but has not been tried.
+    - Need to add routes to machines and/or routers for all Pod IPs.
+
+You need to select an address range for the virtual IPs. 
+- Various approaches:
+  - GCE: each project has own `10.0.0.0/8`.  Carve off a `/16` from that.  Room for several clusters in there.
+  - AWS: use one VPC for whole organization, carve off a chunk for each cluster.  Or use different VPC for different clusters.
+  - IPv6 not supported yet.
+- Allocate one CIDR for PodIPs for each node, or a large CIDR from which
+  smaller CIDRs are automatically allocated to each node (if nodes are dynamically added).
+  - You need Max-pods-per-node * max-number-of-nodes-expected IPs.
+  - e.g. use 10.240.0.0/16
+  - Need to make these routable or connect with overlay.
+- Allocate a subnet for services.  
+  - Does not need to be routable or known to overlay; kube-proxy will translate these to
+    PodIPs.
+  - Call this `SERVICE_CLUSTER_IP_RANGE`
+- Pick a static IP for master node.
+  - Call this the `MASTER_IP`
+- Open any firewalls which would hinder communication between the nodes.
+- Open any firewalls to allow access to the apiserver ports 80 and/or 443.
+
+### Other preparations
```

Comment:
I think we can remove this (once the "Designing and Preparing" section header is added and section TOC inserted)